secureapplication_connector.py

```diff
@@ -8,138 +6,923 @@
-                    error_message = f"Error from server. Status Code: {response.status_code}. Error: {resp_json.get('error', 'Unknown error')}"
-                except ValueError:
-                    error_message = f"Error from server. Status Code: {response.status_code}. Error: {response.text}"
-
-            return action_result.set_status(phantom.APP_ERROR, error_message), None
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, f"Invalid method: {method}"), resp_json)
+
+        # Create a URL to connect to
+        url = self._base_url + endpoint
 
+        try:
+            r = request_func(url, verify=config.get("verify_server_cert", False), **kwargs)
         except Exception as e:
-            return action_result.set_status(phantom.APP_ERROR, f"Error making REST call: {e!s}"), None
+            return RetVal(action_result.set_status(phantom.APP_ERROR, f"Error Connecting to server. Details: {e!s}"), resp_json)
+
+        # Handle 204 No Content
+        if r.status_code == 204:
+            return RetVal(action_result.set_status(phantom.APP_SUCCESS), {})
+
+        return self._process_response(r, action_result)
 
     def _handle_test_connectivity(self, param):
-        """
-        Validate the asset configuration for connectivity using supplied credentials.
-        """
+        self.save_progress(f"In action handler for: {self.get_action_identifier()}")
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        endpoint = "/controller/argento/public-api/v1/libraries"
+        headers = self._get_rest_api_headers(token=self._token)
+
+        # make REST call - list all libaries
+        ret_val, response = self._make_rest_call(endpoint, action_result, headers=headers, method="get")
+        if self._debug:
+            self.debug_print(f"Test connectivity response: {response}")
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Test connectivity successful")
+
+    def _handle_create_new_policy(self, param):
+        self.save_progress(f"In action handler for: {self.get_action_identifier()}")
         action_result = self.add_action_result(ActionResult(dict(param)))
-        self.save_progress("Connecting to instance...")
 
-        # Example test connectivity code
-        endpoint = SECUREAPPLICATION_CONNECTIVITY_ENDPOINT
-        headers = {"Authorization": f"Bearer {self._api_key}"}
+        required_params = ["type", "application_id", "tier_id", "default_action", "enable_policy"]
+        missing = []
+
+        for p in required_params:
+            if not param.get(p):
+                missing.append(p)
+
+        if missing:
+            missing_str = ", ".join(missing)
+            return action_result.set_status(phantom.APP_ERROR, "Missing required parameter(s): " + missing_str)
```

Comment:
Use format string when you can (bit cleaner)
```suggestion
return action_result.set_status(phantom.APP_ERROR, f"Missing required parameter(s): {missing_str}")
```

Reply:
Try to apply this everywhere 👍

Reply:
done. thanks for the suggestion. Had missed doing this at a couple places